bee-tangle/src/unconfirmed_message.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_common::packable::{Packable, Read, Write};
+use bee_message::MessageId;
+
+use std::ops::Deref;
+
+/// A type representing an unconfirmed message.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct UnconfirmedMessage(MessageId);
```

Comment:
`#[repr(transparent)]`

Reply:
Why? `repr(transparent)` has a very specific meaning that does not apply here.

Reply:
Isn't the only reason we have this wrapper to satisfy the type system? I see no reason why this can't have the same memory layout as a MessageId.

Reply:
`repr(transparent)` is used when you want to *guarantee* that the representations are the same for a specific reason (i.e: you're unsafely transmuting between types). If the compiler chooses a different representation to the one you expect, it's because it knows something that you don't and we should let it choose that representation if it so desires. `repr(transparent)` just provides the compiler with fewer options to generate efficient code and should be avoided unless there's a specific reason for it.

Reply:
Alright, you're the expert on this, so I leave it up to you. But as someone who has worked with this type in the last couple of days quite a lot I know that this type is nothing but a `MessageId`, so maybe in the future it becomes important that we can treat is as such. That is what I suggested to tell the compiler. However, it's your choice.